### ISSUE TITLE
[Suggestion] Fighting Mouse-Tracking

### ISSUE THREAD

**Issue by Kraxys** [sereneblue/chameleon#541]
- [ x] I am running the latest version
- [x ] I checked the documentation and found no answer
- [x ] I checked to make sure that this issue has not already been filed

I just found some references relative to mouse tracking.

As written [here](https://wwwen.uni.lu/university/news/latest_news/mouse_movements_reveal_your_behaviour):

> (...) mouse tracking has very practical applications for webmasters, and in particular for search engines. Dr. Ioannis Aparakis from Telefonica Research and co-author of both publications, clarifies: “When you search for something at Google or Bing, your mouse movements are sending a tiny signal to the search engine indicating if you are interested or not in the content you have been shown. As mouse tracking may have privacy issues, we investigated the possibility of recording only a small part of the whole movement trajectory and see if we can still infer how people make choices in web search.

A researcher coded 2 years ago a Chrome extension named [MouseFaker](https://github.com/luileito/mousefaker), available on Github (and seemingly abandonned now) in order to "_prevent user profiling via mouse tracking by implementing the adversarial noise technique described_" in a previous scientific article.

I failed to test it on any Chromium based browser because the actions required in Mouse Faker code were not compatible with manifest v3. Very plausibly this extension works in the background. 

Could this kind of code be ported to Chameleon in order to offer some level of protection against this kind of tracking?

**Comment by sereneblue** [comment]
Yes, this code could be ported to Chameleon. However, Manifest V2 will eventually be deprecated. I intend to migrate Chameleon to Manifest V3 but I'm not sure yet what that version of Chameleon will look like.

**Comment by Kraxys** [comment]
@sereneblue:
Manifest v3 seems to be a burden for many very useful extensions,. Mozilla [wrote](https://blog.mozilla.org/addons/2021/05/27/manifest-v3-update/) that "_Manifest v2 but expect it to be supported for at least one year after Manifest v3 becomes stable in the release_". How much time do you think that leaves?

From my point of view, the Chameleon's core feature are UA, Timezone and Language spoofing. And few months ago, a Chrome **V3 compatible**  extension appeared ([Vytal](https://github.com/z0ccc/Vytal)), aiming to" timezone, locale, geolocation and user agent" (Chameleon doesn't spoof geolocation yet). For that, this extension uses the chrome‧debugger API, Vytal's dev claiming that using this API 
"_allows the data to be spoofed in frames, web workers and during the initial loading of a website. It also makes the spoofing **completely undetectable**_".

The sad thing is that FF doesn't support this chrome‧debugger API yet. But maybe, at least for compatibility reasons with Chrome, Mozilla will make this API available on FF in a not to long future?? Maybe extension's dev could ask Mozilla for that?

**Comment by sereneblue** [comment]
@Kraxys Manifest v2 will likely stick around in Firefox until late 2023. I'll post relevant Manifest v3 updates in the wiki/pin an issue if there's some functionality that can't be ported over once I begin development.

Firefox's debugging tools are a bit limited compared to what Chrome offers so I don't think Mozilla would make an API for it.

**Comment by coolst3r** [comment]
> Yes, this code could be ported to Chameleon. However, Manifest V2 will eventually be deprecated. I intend to migrate Chameleon to Manifest V3 but I'm not sure yet what that version of Chameleon will look like.

can i test it for you v3 version

**Comment by sereneblue** [comment]
> > Yes, this code could be ported to Chameleon. However, Manifest V2 will eventually be deprecated. I intend to migrate Chameleon to Manifest V3 but I'm not sure yet what that version of Chameleon will look like.
> 
> can i test it for you v3 version

Sure, thanks for offering. I will pin an issue for it when the beta version is ready. There doesn't seem to be a definite timeline for V3 now, so it'll be a while before I have a build available.
